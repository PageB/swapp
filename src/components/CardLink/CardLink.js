import React from 'react';
import PropTypes from 'prop-types';

import styles from './CardLink.module.scss';

const cardLink = props => {
  return (
    <div
      className={[styles.Card, styles[props.theme]].join(' ')}
      onClick={() => props.navigate(props.card)}
    >
      <img className={styles.Card__Image} src={props.card.image} alt={props.card.name} />
      <div className={styles.Card__Title}>
        <p className={styles.Card__Text}>{props.card.name}</p>
      </div>
    </div>
  );
};

cardLink.propTypes = {
  theme: PropTypes.oneOf(['DarkTheme', 'LightTheme']),
};

cardLink.defaultProps = {
  theme: 'DarkTheme',
};

export default cardLink;
