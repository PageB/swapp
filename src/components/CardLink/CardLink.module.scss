@import '../../colors.scss';
@import '../../constants.scss';

.Card {
  display: flex;
  flex-direction: row;
  cursor: pointer;
  width: $card-width;
  margin: 8px 16px;
  background-color: $dark-charcoal;
  color: $vivid-yellow;
  border-radius: 8px;

  &__Image {
    border-radius: 8px 0 0 8px;
    width: 60px;
  }

  &__Title {
    color: inherit;
    background-color: inherit;
    padding: 0 16px;
  }

  &__Text {
    color: inherit;
    font-weight: bold;
    font-size: 20px;
    border-radius: 8px 8px 0 0;
  }
}

.DarkTheme {
  background-color: $dark-charcoal;
  color: $vivid-yellow;
}

.LightTheme {
  background-color: $white;
  color: $crayola;
}
