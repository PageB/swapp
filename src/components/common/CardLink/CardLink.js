import React from 'react';

import styles from './CardLink.module.scss';

const cardLink = props => {
  return (
    <div
      className={[styles.Card, styles[props.theme]].join(' ')}
      onClick={() => props.navigate(props.card)}
    >
      <img className={styles.Card__Image} src={props.card.image} alt={props.card.name} />
      <div className={styles.Card__Title}>
        <p className={styles.Card__Text}>{props.card.name}</p>
      </div>
    </div>
  );
};

export default cardLink;
