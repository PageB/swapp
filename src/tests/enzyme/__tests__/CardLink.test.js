import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import { shallow, configure } from 'enzyme';
import { starWarsCharacters } from '../../../mock/data';

import CardLink from '../../../components/CardLink/CardLink';

configure({ adapter: new Adapter() });

describe('CardLink Component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CardLink card={starWarsCharacters[0].starships[0]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Calls closure when card is clicked', () => {
    const onCardClick = sinon.spy();
    const wrapper = shallow(
      <CardLink card={starWarsCharacters[0].starships[0]} navigate={onCardClick} />,
    );

    wrapper.find('.Card').simulate('click');

    expect(onCardClick.calledOnce).toBe(true);
  });
});
