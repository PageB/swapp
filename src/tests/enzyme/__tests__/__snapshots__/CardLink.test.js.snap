// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardLink Component should match snapshot 1`] = `
<div
  className="Card DarkTheme"
  onClick={[Function]}
>
  <img
    alt="Jedi Starfighter"
    className="Card__Image"
    src="https://www.super-hobby.com/zdjecia/7/1/9/20828_rd.jpg"
  />
  <div
    className="Card__Title"
  >
    <p
      className="Card__Text"
    >
      Jedi Starfighter
    </p>
  </div>
</div>
`;
